pkg/sql/show_create.go

```diff
@@ -201,14 +201,22 @@ func ShowCreateTable(
 			}
 			f.WriteString(foreignKey.String())
 		}
-		if idx.ID != desc.PrimaryIndex.ID {
+		// Only add indexes to the create_statement column, and not to the
+		// create_nofks column if they are not associated with an INTERLEAVE
+		// statement.
+		if idx.ID != desc.PrimaryIndex.ID && (len(idx.Interleave.Ancestors) == 0 || ignoreFKs != IgnoreFKs) {
```

Comment:
maybe `if interleave := len(idx.Interleave.Ancestors) == 0 || ignoreFKs != IgnoreFKs; idx.ID != desc.PrimaryIndex.ID && interleave {` then you can use `interleave` below instead of redoing the computation.